cpp/daal/src/data_management/finiteness_checker_cpu.cpp

```diff
@@ -0,0 +1,917 @@
+            const float * colPtr = colBlock.get();
+            sum += static_cast<double>(computeSum<float, cpu>(1, nRows, &colPtr));
+            break;
+        }
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+        {
+            ReadColumns<double, cpu> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(colBlock);
+            const double * colPtr = colBlock.get();
+            sum += computeSum<double, cpu>(1, nRows, &colPtr);
+            break;
+        }
+        default: break;
+        }
+        sumIsFinite &= !valuesAreNotFinite(&sum, 1, false);
+    }
+
+    return sum;
+}
+
+template <typename DataType, daal::CpuType cpu>
+bool checkFiniteness(const size_t nElements, size_t nDataPtrs, size_t nElementsPerPtr, const DataType ** dataPtrs, bool allowNaN)
+{
+    bool notFinite = false;
+    for (size_t ptrIdx = 0; ptrIdx < nDataPtrs; ++ptrIdx) notFinite = notFinite || valuesAreNotFinite(dataPtrs[ptrIdx], nElementsPerPtr, allowNaN);
+
+    return !notFinite;
+}
+
+template <daal::CpuType cpu>
+bool checkFinitenessSOA(NumericTable & table, bool allowNaN, services::Status & st)
+{
+    bool valuesAreFinite                        = true;
+    const size_t nRows                          = table.getNumberOfRows();
+    const size_t nCols                          = table.getNumberOfColumns();
+    NumericTableDictionaryPtr tableFeaturesDict = table.getDictionarySharedPtr();
+
+    for (size_t i = 0; (i < nCols) && valuesAreFinite; ++i)
+    {
+        switch ((*tableFeaturesDict)[i].getIndexType())
+        {
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT32:
+        {
+            ReadColumns<float, cpu> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(colBlock);
+            const float * colPtr = colBlock.get();
+            valuesAreFinite &= checkFiniteness<float, cpu>(nRows, 1, nRows, &colPtr, allowNaN);
+            break;
+        }
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+        {
+            ReadColumns<double, cpu> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(colBlock);
+            const double * colPtr = colBlock.get();
+            valuesAreFinite &= checkFiniteness<double, cpu>(nRows, 1, nRows, &colPtr, allowNaN);
+            break;
+        }
+        default: break;
+        }
+    }
+
+    return valuesAreFinite;
+}
+
+#if defined(DAAL_INTEL_CPP_COMPILER)
+
+const size_t BLOCK_SIZE       = 8192;
+const size_t THREADING_BORDER = 262144;
+
+template <typename DataType>
+DataType getInf()
+{
+    DataType inf;
+    if (sizeof(DataType) == 4)
+        *((uint32_t *)(&inf)) = floatExpMask;
+    else
+        *((uint64_t *)(&inf)) = doubleExpMask;
+    return inf;
+}
+
+    #if defined(__AVX2__) and !defined(__AVX512F__)
+
+template <typename DataType>
+DataType sumWithAVX2(size_t n, const DataType * dataPtr)
+{
+    const size_t nPerInstr = 32 / sizeof(DataType);
```

Comment:
Those calculations are not obvious.
It's better to have:
```suggestion
    constexpr size_t avx2RegisterLength = 256;
    constexpr size_t numberOfBitsInByte = 8;
    constexpr size_t nPerInstr = avx2RegisterLength / (numberOfBitsInByte * sizeof(DataType));
```